datadog/fwprovider/resource_datadog_apm_retention_filter_order.go

```diff
@@ -0,0 +1,187 @@
+	FilterIds []types.String `tfsdk:"filter_ids"`
+}
+
+func NewApmRetentionFiltersOrderResource() resource.Resource {
+	return &ApmRetentionFiltersOrderResource{}
+}
+func (r *ApmRetentionFiltersOrderResource) Configure(_ context.Context, request resource.ConfigureRequest, response *resource.ConfigureResponse) {
+	providerData := request.ProviderData.(*FrameworkProvider)
+	r.Api = providerData.DatadogApiInstances.GetApmRetentionFiltersApiV2()
+	r.Auth = providerData.Auth
+}
+
+func (r *ApmRetentionFiltersOrderResource) Metadata(_ context.Context, request resource.MetadataRequest, response *resource.MetadataResponse) {
+	response.TypeName = "apm_retention_filter_order"
+}
+func (d *ApmRetentionFiltersOrderResource) Schema(_ context.Context, _ resource.SchemaRequest, response *resource.SchemaResponse) {
+	response.Schema = schema.Schema{
+		Description: "Provides a Datadog [APM Retention Filters API](https://docs.datadoghq.com/api/v2/apm-retention-filters/) resource, which is used to manage Datadog APM retention filters order.",
+		Attributes: map[string]schema.Attribute{
+			"filter_ids": schema.ListAttribute{
+				Description: "The filter IDs list. The order of filters IDs in this attribute defines the overall APM retention filters order. If `filter_ids` is empty or not specified, it will import the actual order, and create the resource. Otherwise, it will try to update the order.",
+				ElementType: types.StringType,
+				Required:    true,
+			},
+			"id": utils.ResourceIDAttribute(),
+		}}
+}
+
+func (r *ApmRetentionFiltersOrderResource) ImportState(ctx context.Context, request resource.ImportStateRequest, response *resource.ImportStateResponse) {
+	resource.ImportStatePassthroughID(ctx, frameworkPath.Root("id"), request, response)
+}
+
+func (r *ApmRetentionFiltersOrderResource) Read(ctx context.Context, request resource.ReadRequest, response *resource.ReadResponse) {
+	var state ApmRetentionFiltersOrderModel
+	response.Diagnostics.Append(request.State.Get(ctx, &state)...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	resp, httpResp, err := r.Api.ListApmRetentionFilters(r.Auth)
+	if err != nil {
+		if httpResp != nil && httpResp.StatusCode == 404 {
+			response.State.RemoveResource(ctx)
+			return
+		}
+		response.Diagnostics.Append(utils.FrameworkErrorDiag(err, "error retrieving retention filter"))
+		return
+	}
+	if err := utils.CheckForUnparsed(resp); err != nil {
+		response.Diagnostics.AddError("response contains unparsedObject", err.Error())
+		return
+	}
+
+	r.updateState(ctx, &state, &resp)
+
+	// Save data into Terraform state
+	response.Diagnostics.Append(response.State.Set(ctx, &state)...)
+}
+
+func (r *ApmRetentionFiltersOrderResource) Create(ctx context.Context, request resource.CreateRequest, response *resource.CreateResponse) {
+	var state ApmRetentionFiltersOrderModel
+	response.Diagnostics.Append(request.Plan.Get(ctx, &state)...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	if len(state.FilterIds) > 0 {
```

Comment:
Do we need this conditional since we marked the field as required?

Reply:
no, I think it should be fine with required.

Reply:
I removed it